rigs/heft-node-rig/profiles/default/config/heft.json

```diff
@@ -31,6 +31,12 @@
           "taskPlugin": {
             "pluginPackage": "@rushstack/heft-api-extractor-plugin"
           }
+        },
+        "node-service": {
+          "taskDependencies": ["typescript"],
+          "taskEvent": {
+            "eventKind": "nodeService"
+          }
         }
```

Comment:
Prior to Heft 0.51.0, `NodeServicePlugin` was always enabled, so what I'm fixing here is arguably a regression.  If you look at `NodeServicePlugin.ts`, all the logic is skipped unless the `--serve` parameter is provided, so the perf concern should be pretty negligible.

We probably should separate `heft-node-rig` into two profiles (`app` and `library` instead of `default`), and then `library` could omit this plugin, however that change is separate from this PR.

Reply:
Agree with splitting the node rig into the two profiles. Admittedly I'm not a huge fan of having this be present in the normal rig, especially since it'll rarely get used.